app/controllers/BankDetailsController.scala

```diff
@@ -118,29 +119,41 @@ class BankDetailsController @Inject() (
-            Ok(views.enterBankDetailsPage(
-              form.withError(error),
-              BankDetailsController.chooseTypeOfAccountUrl
-            ))
-          }
-
-        journeyService.updateDirectDebitDetails(request.journeyId, directDebitDetails)
-          .flatMap { _ =>
-            bankDetailsForm.isSoleSignatory match {
-              case IsSoleSignatoryFormValue.No =>
-                Future.successful(Redirect(routes.BankDetailsController.cannotSetupDirectDebitOnlinePage))
-
-              case IsSoleSignatoryFormValue.Yes =>
-                barsService.assessBankAccountReputation(directDebitDetails.bankDetails).map {
-                  case accountNumberIsWellFormattedNo() => enterBankDetailsPageWithBarsError(accountNumberNotWellFormatted)
-                  case sortCodeSupportsDirectDebitNo()  => enterBankDetailsPageWithBarsError(sortCodeDoesNotSupportsDirectDebit)
-                  case sortCodeIsPresentOnEiscdNo()     => enterBankDetailsPageWithBarsError(sortCodeNotPresentOnEiscd)
-                  // BARs check was successful
-                  case _                                => Redirect(routes.BankDetailsController.checkBankDetails)
-                }
+        // format: OFF
+        def enterBankDetailsPageWithBarsError(error: FormError): Result = {
+          Ok(views.enterBankDetailsPage(
+            form = form.withError(error),
+            backUrl = BankDetailsController.chooseTypeOfAccountUrl
+          ))
+        }
+        // format: ON
+
+        request.journey match {
+          case j: AfterChosenTypeOfBankAccount =>
+            for {
+              barsResponseOpt <- barsService.verifyBankDetails(directDebitDetails.bankDetails, bankDetailsForm.isSoleSignatory, j.typeOfBankAccount)
+              // TODO update backend + lockout logic
+              _ <- journeyService.updateDirectDebitDetails(request.journeyId, directDebitDetails)
+            } yield {
+              (barsResponseOpt, bankDetailsForm.isSoleSignatory) match {
+                case (_, IsSoleSignatoryFormValue.No) =>
+                  Redirect(routes.BankDetailsController.cannotSetupDirectDebitOnlinePage)
+
+                // TODO add helper (for below?)
+                case (Some(thirdPartyError()), _) => Redirect(routes.BankDetailsController.barsErrorPlaceholder)
+
+                case (Some(accountNumberIsWellFormattedNo()), _) => enterBankDetailsPageWithBarsError(accountNumberNotWellFormatted)
+                case (Some(sortCodeSupportsDirectDebitNo()), _) => enterBankDetailsPageWithBarsError(sortCodeDoesNotSupportsDirectDebit)
+                case (Some(sortCodeIsPresentOnEiscdNo()), _) => enterBankDetailsPageWithBarsError(sortCodeNotPresentOnEiscd)
+
+                case (Some(nameMatchesNo()), _) => enterBankDetailsPageWithBarsError(nameDoesNotMatch)
+
+                case _ => Redirect(routes.BankDetailsController.checkBankDetails)
+              }
             }
-          }
+          case _ =>
```

Comment:
I think we're trying to avoid using wildcards, should be covered if you have a `case j: BeforeChosenTypeOfBankAccount => Future.successful(JourneyIncorrectStateRouter.logErrorAndRouteToDefaultPage(request.journey))`  on line 132

Reply:
ah yes 👍

Reply:
will fix that - good spot